src/main/java/seedu/duke/TaskList.java

```diff
@@ -0,0 +1,102 @@
+package seedu.duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a list of tasks, and provide methods for user to modify this list of tasks.
+ */
+public class TaskList {
+    private List<Task> listOfTasks;
+
+    /**
+     * Initializes an empty list of tasks.
+     */
+    public TaskList() {
+        this.listOfTasks = new ArrayList<>();
+    }
+
+    /**
+     * Initializes the list of tasks with a pre-existing list.
+     * @param listOfTasks The pre-existing list of tasks.
+     */
+    public TaskList(List<Task> listOfTasks) {
+        this.listOfTasks = listOfTasks;
+    }
+
+    /**
+     * Returns the list of tasks as a List.
+     * @return List of tasks.
+     */
+    public List<Task> getListOfTasks() {
+        return this.listOfTasks;
+    }
+
+    /**
+     * Returns the deadlines and events filtered by a particular date.
+     * @param date The date used to filter the tasks.
+     * @return List of deadlines and events occurring on the date.
+     */
+    public List<Task> getListOfTasks(LocalDate date) {
+        return this.listOfTasks.stream().filter(task -> (task instanceof Event && ((Event) task).isOnDate(date))
+                || (task instanceof Deadline && ((Deadline) task).isOnDate(date))).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of tasks which description contains the keyword.
+     * @param keyword The keyword to be used to filter the tasks.
+     * @return The list of tasks.
+     */
+    public List<Task> getListOfTasks(String keyword) {
```

Comment:
Maybe it is better to name this as `getTaskListWithKeyword` instead of an overloaded method so that the functionality of the method can be easily understood without reading the docs? 🤔